conn.go

```diff
@@ -421,6 +449,122 @@ func decodeXtext(val string) (string, error) {
 	return decoded, nil
 }
 
+// This regexp matches 'EmbeddedUnicodeChar' token defined in
+// https://datatracker.ietf.org/doc/html/rfc6533.html#section-3
+// however it is intentionally relaxed by requiring only '\x{HEX}' to be
+// present.  It also matches disallowed characters in QCHAR and QUCHAR defined
+// in above.
+// So it allows us to detect malformed values and report them appropriately.
+var eUOrDCharRe = regexp.MustCompile(`\\x[{][0-9A-F]+[}]|[[:cntrl:] \\+=]`)
```

Comment:
In general I'm not a fan of using regexp for decoding but we already do that in go-smtp so it's not a big deal…

Reply:
I'm not a fan too, but could't think of a better way.